src/handler/http/models/destinations.rs

```diff
@@ -0,0 +1,241 @@
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! These models define the schemas of HTTP request and response JSON bodies in
+//! destinations and templates API endpoints.
+
+use std::fmt;
+
+use config::{ider, meta::destinations as meta_dest};
+use hashbrown::HashMap;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::service::db::alerts::destinations::DestinationError;
+
+impl From<meta_dest::Destination> for Destination {
+    fn from(value: meta_dest::Destination) -> Self {
+        match value.module {
+            meta_dest::Module::Alert {
+                template,
+                destination_type,
+            } => match destination_type {
+                meta_dest::DestinationType::Email(email) => Self {
+                    name: value.name,
+                    emails: email.recipients,
+                    template: Some(template),
+                    destination_type: DestinationType::Email,
+                    ..Default::default()
+                },
+                meta_dest::DestinationType::Http(endpoint) => Self {
+                    name: value.name,
+                    url: endpoint.url,
+                    method: endpoint.method,
+                    skip_tls_verify: endpoint.skip_tls_verify,
+                    headers: endpoint.headers,
+                    destination_type: DestinationType::Http,
+                    template: Some(template),
+                    ..Default::default()
+                },
+                meta_dest::DestinationType::Sns(aws_sns) => Self {
+                    name: value.name,
+                    template: Some(template),
+                    sns_topic_arn: Some(aws_sns.sns_topic_arn),
+                    aws_region: Some(aws_sns.aws_region),
+                    destination_type: DestinationType::Sns,
+                    ..Default::default()
+                },
+            },
+            meta_dest::Module::Pipeline {
+                pipeline_id,
+                endpoint,
+            } => Self {
+                name: value.name,
+                url: endpoint.url,
+                method: endpoint.method,
+                skip_tls_verify: endpoint.skip_tls_verify,
+                headers: endpoint.headers,
+                pipeline_id: Some(pipeline_id),
+                destination_type: DestinationType::Http,
+                ..Default::default()
+            },
+        }
+    }
+}
+
+impl Destination {
+    pub fn into(self, org_id: &str) -> Result<meta_dest::Destination, DestinationError> {
+        let id = String::new(); // placeholder, set at db layer
```

Comment:
This is related to my other comments about using `Option<svix_ksuid::Ksuid>` instead of `String` for the type of `id`:

If we use `Option<svix_ksuid::Ksuid>` here instead of `String` then we can set this to `None`.